pkg/inject/oauth-proxy.go

```diff
@@ -0,0 +1,58 @@
+package inject
+
+import (
+	"fmt"
+
+	"github.com/spf13/viper"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+
+	"github.com/jaegertracing/jaeger-operator/pkg/account"
+	"github.com/jaegertracing/jaeger-operator/pkg/apis/io/v1alpha1"
+	"github.com/jaegertracing/jaeger-operator/pkg/service"
+)
+
+// OAuthProxy injects an appropriate OpenShift proxy into the given deployment
+func OAuthProxy(jaeger *v1alpha1.Jaeger, dep *appsv1.Deployment) *appsv1.Deployment {
+	if jaeger.Spec.Ingress.OAuthProxy == nil || !*jaeger.Spec.Ingress.OAuthProxy {
```

Comment:
Shouldn't this also check for platform? All-in-one and production controllers don't check for platform before calling.

Reply:
The `normalize()` operation will set the flag to `false` if the platform doesn't support the OAuth proxy. I should actually remove the reference to OpenShift from this `godoc`.